azurerm/internal/services/compute/resource_arm_disk_encryption_set.go

```diff
@@ -0,0 +1,387 @@
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validate.NoEmptyStrings,
+						},
+					},
+				},
+			},
+
+			"tags": tags.Schema(),
+		},
+	}
+}
+
+func resourceArmDiskEncryptionSetCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Compute.DiskEncryptionSetsClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	if features.ShouldResourcesBeImported() && d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroup, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for present of existing Disk Encryption Set %q (Resource Group %q): %+v", name, resourceGroup, err)
+			}
+		}
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_disk_encryption_set", *existing.ID)
+		}
+	}
+
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	activeKey := d.Get("active_key").([]interface{})
+	t := d.Get("tags").(map[string]interface{})
+
+	diskEncryptionSet := compute.DiskEncryptionSet{
+		Location: utils.String(location),
+		EncryptionSetProperties: &compute.EncryptionSetProperties{
+			ActiveKey: expandArmDiskEncryptionSetKeyVaultAndKeyReference(activeKey),
+		},
+		Tags: tags.Expand(t),
+	}
+
+	if v, ok := d.GetOk("identity"); ok {
+		diskEncryptionSet.Identity = expandArmDiskEncryptionSetIdentity(v.([]interface{}))
+	} else {
+		diskEncryptionSet.Identity = &compute.EncryptionSetIdentity{
+			Type: compute.SystemAssigned,
+		}
+	}
+
+	// validate whether the keyvault has soft-delete and purge-protection enabled
+	err := validateKeyVaultAndKey(ctx, meta, resourceGroup, diskEncryptionSet.ActiveKey)
+	if err != nil {
+		return fmt.Errorf("Error creating Disk Encryption Set %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	future, err := client.CreateOrUpdate(ctx, resourceGroup, name, diskEncryptionSet)
+	if err != nil {
+		return fmt.Errorf("Error creating Disk Encryption Set %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for creation of Disk Encryption Set %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		return fmt.Errorf("Error retrieving Disk Encryption Set %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+	if resp.ID == nil {
+		return fmt.Errorf("Cannot read Disk Encryption Set %q (Resource Group %q) ID", name, resourceGroup)
+	}
+	d.SetId(*resp.ID)
+
+	return resourceArmDiskEncryptionSetRead(d, meta)
+}
+
+func validateKeyVaultAndKey(ctx context.Context, meta interface{}, resourceGroup string, keyVaultAndKey *compute.KeyVaultAndKeyReference) error {
```

Comment:
rather than passing in meta here can you pass in the specific clients needed here?

Reply:
Sure, now I am passing *client,Client to this method